frontend/console/src/features/modules/schema/schema.utils.ts

```diff
@@ -28,3 +71,56 @@ export function isFirstLineOfBlock(ll: string[], i: number): boolean {
   }
   return staticKeywords.includes(tokens[0]) || declTypes.includes(tokens[0])
 }
+
+export interface DeclSchema {
+  schema: string
+  declType: string
+}
+
+export function declFromModules(moduleName: string, declName: string, modules: UseQueryResult<GetModulesResponse, Error>) {
+  if (!modules.isSuccess || modules.data.modules.length === 0) {
+    return
+  }
+  const module = modules.data.modules.find((module) => module.name === moduleName)
+  if (!module?.schema) {
+    return
+  }
+  return declFromModuleSchemaString(declName, module.schema)
+}
+
+export function declFromModuleSchemaString(declName: string, schema: string) {
+  const lines = schema.split('\n')
```

Comment:
A lot of this function and file seems to be scanning the schema string to pull it back apart into decls, but we can get the raw schema as well and not have to deal with regexing on the string if that's easier.

Reply:
posting update from our slack discussion: we'll do that with the new endpoint: https://github.com/TBD54566975/ftl/issues/2805